classes/controller/edit_controller.php

```diff
@@ -212,27 +212,29 @@ public function update_post_action() {
             $post       = $DB->get_record('hsuforum_posts', array('id' => $postid), '*', MUST_EXIST);
             $discussion = $DB->get_record('hsuforum_discussions', array('id' => $post->discussion, 'forum' => $forum->id), '*', MUST_EXIST);
 
-            if (empty($groupid)) {
-                $groupid = -1;
+            if (empty($groupids)) {
+                $groupids[] = -1;
             }
+
             // If private reply, then map it to the parent author user ID.
             if (!empty($privatereply)) {
                 $parent     = $DB->get_record('hsuforum_posts', array('id' => $post->parent), '*', MUST_EXIST);
                 $privatereply = $parent->userid;
             }
+
             return $this->postservice->handle_update_post($course, $cm, $forum, $context, $discussion, $post, $files, $posttomygroups,
-              array(
+                array (
                 'subject'       => $subject,
                 'name'          => $subject,
-                'groupid'       => $groupid,
+                'groupids'      => $groupids,
                 'itemid'        => $itemid,
                 'message'       => $message,
                 'messageformat' => $messageformat,
                 'reveal'        => $reveal,
                 'privatereply'  => $privatereply,
                 'timestart'     => $timestart,
                 'timeend'       => $timeend
-            ));
+                ));
```

Comment:
Is this indentation correct?

Reply:
![Screen Shot 2022-04-01 at 1 57 02 PM](https://user-images.githubusercontent.com/71312749/161259096-b8e2eb15-6d7f-4e4f-994c-d6e9d08c0ee9.png)
![Screen Shot 2022-04-01 at 1 54 54 PM](https://user-images.githubusercontent.com/71312749/161259111-0e171241-4a0e-4132-8b06-605cda6baecc.png)
![Screen Shot 2022-04-01 at 1 44 46 PM](https://user-images.githubusercontent.com/71312749/161259113-e54c12ca-1d06-4dde-a8fe-7bbf2f64b286.png)